src/classes/ALLO_Allocations_TEST.cls

```diff
@@ -124,6 +124,71 @@ private with sharing class ALLO_Allocations_TEST {
         system.assertEquals(8, queryOpps[0].Amount, 'The opportunity amount should remain unchanged after the exception is caught.');
     }
 
+
+    /*******************************************************************************************************
+    * @description When changing the amount of a Payment with Allocations: 
+    * Increasing the amount will adjust the percentage allocations.
+    * Decreasing the amount so that the total allocation amount is below the payment amount will adjust 
+    * the percentage allocations.
+    * Decreasing the amount so that the total allocation amount is greater than the payment amount, an 
+    * error is thrown and the payment update is blocked.
+    ********************************************************************************************************/
+    static testMethod void pmtAmountChange() {
```

Comment:
Hey @RadSFDeveloper, I think we already talked about this in the Best Practices doc. In a different platform I'd totally agree with you that we should test only one scenario per test. In Salesforce however, I'd try to cover as many scenarios as possible in one test. The reason being how long the build takes to run (with 60-70% of the time consumed by tests running).

Reply:
I generally agree with @ceiroa on this, though - and not part of something I think we should refactor for at this point - in an ideal world a unit test class would have a _@TestSetup_ method that generates all the test data at once so that the class can have individual unit test methods to test each scenario without negatively impacting performance. These older unit test classes are a train wreck. I'm fairly sure we've added a stories to the backlog to start planning a complete unit test refactor.

Reply:
We are aiming to decrease the test execution time by minimizing data requirement for tests. For example, we can write new methods not to rely on data creation and data save, if possible. Thus, new unit tests can be written so they test only one feature per test. Of course, we do need an integration test and data creation for it will prolong test execution. Moreover, we can start with using the `@testSetup` currently. These are just suggestions, and since we are creating new methods and new unit tests, I thought it might apply.  It is ok if the code is left as is.